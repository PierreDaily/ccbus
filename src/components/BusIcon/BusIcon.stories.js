import BusIcon from "./index.vue";

// More on default export: https://storybook.js.org/docs/vue/writing-stories/introduction#default-export
export default {
  title: "BusIcon",
  component: BusIcon,
  // More on argTypes: https://storybook.js.org/docs/vue/api/argtypes
  argTypes: {
    busLetter: { control: "text" },
  },
};

// More on component templates: https://storybook.js.org/docs/vue/writing-stories/introduction#using-args
const Template = (args) => ({
  // Components used in your story `template` are defined in the `components` object
  components: { BusIcon },
  // The story's `args` need to be mapped into the template through the `setup()` method
  setup() {
    return { args };
  },
  // And then the `args` are bound to your component with `v-bind="args"`
  template: '<bus-icon v-bind="args" />',
});

export const Primary = Template.bind({});
// More on args: https://storybook.js.org/docs/vue/writing-stories/args
Primary.args = {
  busLetter: "a",
};
