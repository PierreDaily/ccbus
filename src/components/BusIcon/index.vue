<template>
  <div id ="test" aria-label="B Bus" class="transportation-type">
    <div>
      <img :src="busIcon" />
    </div>
    <div aria-labelledby="test">{{ busLetter?.toUpperCase() || "B" }}</div>
  </div>
</template>

<script lang="ts">
import busIcon from "@/assets/icons/bus.svg";

export default {
  name: "BusIcon",
  props: {
    busLetter: String,
  },
  setup() {
    return {
      busIcon,
    };
  },
};
</script>

<style lang="scss" scoped>
.transportation-type {
  border: solid var(--border-color) 1px;
  border-radius: 2px;
  height: 25px;
  width: 50px;
  overflow: hidden;

  & > div {
    align-items: center;
    display: inline-flex;
    justify-content: center;
    width: 50%;
    height: 100%;
  }

  & > div:first-child {
    background-color: var(--bus-icon-green);
    border-right: solid var(--border-color) 1px;
  }

  & > div:nth-child(2) {
    color: var(--font-color-black);
    font-family: var(--font-primary);
    vertical-align: top;
  }
}

img {
  display: block;
  height: 16px;
  width: 16px;
}
</style>
