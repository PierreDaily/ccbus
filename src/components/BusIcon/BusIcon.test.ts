import { mount, shallowMount } from "@vue/test-utils";
import BusIcon from "./index.vue";

test("Mount component", async () => {
  // arrange
  expect(BusIcon).toBeTruthy();

  const wrapper = mount(BusIcon, {
    props: {
    },
  });

  // assert

  expect(wrapper.html()).toMatchSnapshot();
});

test("Should render 'B' when no props is passed on", async () => {
  // arrange
  expect(BusIcon).toBeTruthy();

  const wrapper = mount(BusIcon, {
    props: {
    },
  });

  // assert
  expect(wrapper.text()).toContain("B");

});

test("Should render 'X' when props is passed on", async () => {
  // arrange
  expect(BusIcon).toBeTruthy();

  const wrapper = mount(BusIcon, {
    props: {
      busLetter: "X",
    },
  });

  // assert
  expect(wrapper.text()).toContain("X");

});

test("Should convert the props to uppercase", async () => {
  // arrange
  expect(BusIcon).toBeTruthy();

  const wrapper = mount(BusIcon, {
    props: {
      busLetter: "x",
    },
  });

  // assert
  expect(wrapper.text()).toContain("X");

});