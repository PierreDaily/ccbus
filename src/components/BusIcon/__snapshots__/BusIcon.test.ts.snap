// Vitest Snapshot v1

exports[`Mount component 1`] = `
"<div id=\\"test\\" aria-label=\\"B Bus\\" class=\\"border border-solid border-grey-light h-[25px] w-[50px] rounded-sm overflow-hidden\\">
  <div class=\\"items-center inline-flex justify-center w-1/2 h-full bg-bus-green border-r-grey-light border-r\\"><img src=\\"/src/assets/icons/bus.svg\\" /></div>
  <div class=\\"items-center inline-flex justify-center w-1/2 h-full align-top font-sans text-black-light\\" aria-labelledby=\\"test\\">B</div>
</div>"
`;
